internal/pkg/v2/infrastructure/redis/event.go

```diff
@@ -131,6 +135,101 @@ func deleteEventById(conn redis.Conn, id string) (edgeXerr errors.EdgeX) {
 	return edgeXerr
 }
 
+// AsyncDeleteEvents Handles the deletion of device readings asynchronously. This function is expected to be running
+// in a go-routine and works with the "deleteReadingsByIds" function for better performance.
+func (c *Client) AsyncDeleteEvents() {
```

Comment:
Move to Core Data App layer and have it use the DB Layer to do the deletes.

Reply:
@lenny-intel I thought this should remain in the infrastructure layer, because it is Redis specific mechanism.  If we use a SQL database, we can delete them in one SQL command by specifying "Where".
Also, core-metadata and other services has its own DBClient interface, so they won't call this func.  

After Hanoi, I will find a way to refactor the infrastructure layer.  Currently, the DBClient struct here would contain all the function for all the services using database.  It would be better put the common logic here move the service specific logics to their own package.

Reply:
@cloudxxx8 , see my comments on init.go. This redis specific code points out why these need to be in the App Layer. Redis is currently the only DB for V2.  If we ever implemented a SQL DB we can revisit the best way to deal with the DBClient interface. 

Question, is the async channel the best why to do this with Redis, just because V1 did it that way?
It seems more complicated than needed. Can the Redis layer DeletePushedEvents just have call a one shot go func that does the work rather than having these long running go threads waiting on a channel to do the same async deleting?

Reply:
@cloudxxx8 , @judehung , please focus on my last comment on eliminating the async functions and channels for a short running local Go func.